daemon/create.go

```diff
@@ -14,6 +14,16 @@ import (
 	"github.com/opencontainers/runc/libcontainer/label"
 )
 
+// ContainerCreateAPI takes configs and creates a container, returning
+// the ID of the newly created container.
+func (daemon *Daemon) ContainerCreateAPI(name string, config *runconfig.Config, hostConfig *runconfig.HostConfig, adjustCPUShares bool) (string, []string, error) {
```

Comment:
It requires changes in builder/internals. If a container can be looked up by ID at the time create returns, it should be okay. How does that sound?